### ISSUE TITLE
Drone: truncated guitest logs

### ISSUE THREAD

**Issue by TheOneRing** [owncloud/client#9276]
We recently increase the verbosity of the squishtests in order to better understand issues however drone decides that we don't need all that log.
The logs generated in the gui tests step are truncated.
Can we upload them as an artifact?

![image](https://user-images.githubusercontent.com/200626/145576692-c9904ac4-3544-461b-bd5d-b4e1fb174ff0.png)
Logs only start at line ... 17470

**Comment by TheOneRing** [comment]
If you have a look on the ctests we run, they write the logs to a tmp file and only print them if the test failed.
This could be a more elegant solution.

**Comment by TheOneRing** [comment]
See https://jira.owncloud.com/browse/OCAT-246

**Comment by TheOneRing** [comment]
`squish | tee >(gzip --stdout > log.txt.gz)`

The resulting artifact should then be pushed to the pr.

I think showGuiTestResult should also be pushed as a comment to the pr.

**Comment by TheOneRing** [comment]
Or we move the remaining steps to github actions, they are much better integrated with the pr workflow and don't limit log files.

**Comment by dragotin** [comment]
@individual-it @phil-davis @micbar any ideas how to solve this?

**Comment by xoxys** [comment]
You will IMO still need a public S3 bucket (prepared by us) to upload and link log artifacts. The currently used drone-github-comment plugin used by you in various other repos to e.g. upload test screenshots can not upload files...

**Comment by phil-davis** [comment]
Some time ago I was getting some code together in core to be able to save screenshots of core UI tests. That is another example of saving artifacts (output results) of CI runs, where the result is not a simple and small amount of text. I did some code as POC to push the locally-saved screenshots on the drone agent to persistent storage: PR https://github.com/owncloud/core/pull/38916 - in that case I setup a share on my own personal cloud storage, just so I could see the screenshots.

But for real use, that sort of thing needs to be pushed by the test pipeline from the drone agent to some persistent storage that the pipeline has write access to (via drone secrets or whatever...). @xoxys suggestion is a public S3 bucket. To avoid nefarious people filling it with crud, it would need to have some security protection for write access. But if the artifacts are all from CI of public repos, then there is no problem giving public read access.

And there would need to be some system for purging old artifacts, rather than storing them forever and needing a Petabye of storage next year!

**Comment by individual-it** [comment]
GitHub comments can be the persistent storage. If we ZIP the file as proposed by @TheOneRing we can just upload the ZIP to the GitHub comments and leave it there. It will clutter PRs but that is the only issue. We could also upload it only on failing PRs

**Comment by Talank** [comment]
If we need more insight, and if we comment the log file to the PR then may be instead of commenting just the `showGuiTestResult`, you might want to comment the whole log generated in the json format. The log is available in `/drone/src/test/guiTestReport/results.json`

**Comment by individual-it** [comment]
Not only cannot the used drone plugin upload files (as @xoxys points out) I cannot find a way to upload files to comments via documented github API
dumping the whole JSON into one comment? :scream:

**Comment by Talank** [comment]
Dumping the whole JSON file makes the PR's overview very polluted. IMO.

**Comment by TheOneRing** [comment]
Well if that doesn't work lets keep them on one of our servers.

**Comment by individual-it** [comment]
yes the only way I see is ether placing the log artifacts on our S3 server (trying that in #9406) and cleaning up regularly. @xoxys do we run regular cleanups on the S3 buckets or can we setup that?
Alternative is of-course moving to GitHub Actions, that allows us to store artifacts see https://github.com/actions/upload-artifact

**Comment by xoxys** [comment]
All this was discussed via chat and the internal Jira ticket already. Now, this ticket was closed, and we have shifted the discussion to another issue... Don't use "random" s3 secret endpoints to test and upload stuff. This needs coordination, at least we need to check and prepare the drone secrets for s3 to use, and right now I don't have time for it. We work in the same way as everyone else. Create a ticket in our Jira Project, we add a priority and plan time/resources for it.... 

Yes, all this is possible with S3, and we actively use it in e.g. the owncloud/web repo to upload screenshots from UI tests to a special bucket defined by us. We have applied retention of 14 days to this bucket and the bucket has public read access to access (and optionally link the URL in a PR comment).

**Comment by individual-it** [comment]
@xoxys I wasn't trying to push you in doing something immediately, sorry if I made that impression. I also wasn't aware that we cannot just use the existing endpoints and secrets. I guess I have a bad concision myself that I haven't worked on this issue for so long.
And you are totally right, my comment above with GH actions is useless  because of the decision made in the JIRA discussion.

I've created a new Jira task https://jira.owncloud.com/browse/OCAT-247 as you have requested
